plugins/inventory/now.py

```diff
@@ -442,8 +454,23 @@ def set_hostvars(self, host, record, columns):
                 "Invalid column names: {0}.".format(", ".join(missing))
             )
 
+        show_values = self.get_option("show_values")
+
         for k in columns:
-            self.inventory.set_variable(host, k.replace(".", "_"), record[k])
+            value = record.get(k)
+            if isinstance(value, dict):
+                main_value = value.get("display_value") or value.get("value") or value
```

Comment:
either use display_value or value but like this is dangerous. You imply that you can use the whole dict if display_value or value are not present but I don't think you need this.

The problem with this code is that you hardcoded `sysparam_display_value` to all, which means you'll always get `display_value` and `value` therefore the first and second `or` are useless.
Also, in the `if` block the for is useless because, as you'll always have a value for `display_value`, you only need the `value` value.
Am I right or I'm missing something here?